addOns/exim/src/main/java/org/zaproxy/addon/exim/urls/UrlsImporter.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.exim.urls;
+
+import java.awt.EventQueue;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import org.apache.commons.httpclient.URI;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.history.ExtensionHistory;
+import org.parosproxy.paros.model.HistoryReference;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpSender;
+import org.parosproxy.paros.view.View;
+
+public final class UrlsImporter {
+
+    private static final Logger LOG = LogManager.getLogger(UrlsImporter.class);
+
+    private UrlsImporter() {}
+
+    public static String importUrlFile(File file) {
+        if (file == null) {
+            return "";
+        }
+        BufferedReader in = null;
+        try {
+            if (View.isInitialised()) {
+                // Switch to the output panel, if in GUI mode
+                View.getSingleton().getOutputPanel().setTabFocus();
+            }
+            in = new BufferedReader(new FileReader(file));
+
+            HttpSender sender =
+                    new HttpSender(
+                            Model.getSingleton().getOptionsParam().getConnectionParam(),
+                            true,
+                            HttpSender.MANUAL_REQUEST_INITIATOR);
+
+            String line;
+            StringBuilder outputLine;
+            while ((line = in.readLine()) != null) {
+                if (!line.startsWith("#") && line.trim().length() > 0) {
+                    if (!line.startsWith("http")) {
+                        // ZAP exports urls to a file in which each line starts with the HTTP Method
+                        // (verb)
+                        // followed by a tab, so makes sense to cope with it.
+                        // Otherwise assume complete URLs starting with http(s) scheme.
+                        int tabIdx = line.indexOf("\t");
+                        if (tabIdx > -1) {
+                            line = line.substring(line.indexOf("\t")).trim();
+                        }
```

Comment:
This is no longer true, so I guess this chunk can be removed?

Reply:
Right.